### PR TITLE
Fix tests for function name inference

### LINKED ISSUE #45: Failing unit tests in master
```
Purdy
  ✔ 1) should indent array correctly (23 ms)
  ✔ 2) allows for removal of array index (5 ms)
  ✔ 3) honors indent level (2 ms)
  ✔ 4) prints arguments function (2 ms)
  ✔ 5) should print plain (2 ms)
  ✔ 6) should handle circular references (5 ms)
  ✔ 7) should print a string (1 ms)
  ✔ 8) should print a date (2 ms)
  ✔ 9) should print a number (1 ms)
  ✔ 10) should print null (1 ms)
  ✔ 11) should print undefined (1 ms)
  ✔ 12) should print a regular expression (1 ms)
  ✔ 13) should print a false boolean (2 ms)
  ✔ 14) should print a true boolean (1 ms)
  ✔ 15) should print an empty array without extra indentation (1 ms)
  ✔ 16) should print an empty object without extra indentation (1 ms)
  ✔ 17) should print a more complex object (4 ms)
  ✔ 18) should not print circular after second use (3 ms)
  ✔ 19) shows circular reference with path (2 ms)
  ✔ 20) will keep a path for an object in Hoek format (4 ms)
  ✔ 21) indents object the way it should (1 ms)
  ✔ 22) aligns object the way it should (2 ms)
  ✔ 23) prints directly to console (2 ms)
  errors
    ✔ 24) should display an error (2 ms)
    ✔ 25) should display an error with no error message and property correctly (1 ms)
    ✔ 26) should display an error with properties (2 ms)
    ✔ 27) should display an error with detail (1 ms)
  functions
    ✔ 28) should print constructor name (2 ms)
    ✖29) should print not print common constructor
    ✔ 30) should print a function (1 ms)
    ✖31) should print an anonymous function
    ✖32) should print properties for functions
    ✔ 33) should print properties for functions with name (1 ms)
  symbols
    ✔ 34) prints symbols (1 ms)
    ✔ 35) prints only symbols (1 ms)
    100% coverage
      ✔ 36) should have coverage for having getOwnPropertySymbols (2 ms)
      - 37) should have coverage for not having getOwnPropertySymbols (0 ms)
  depth
    ✔ 38) should handle depth printing (2 ms)
    ✔ 39) should handle depth printing for array (1 ms)
    ✔ 40) should print using zero depth (1 ms)
    ✔ 41) should handle depth printing for mixed (2 ms)
    ✔ 42) should handle depth printing using null depth (1 ms)
```

### PR DESCRIPTION
**Pull request by mightyiam**
(fixes #45)

### CODE REVIEW

**Comment by danielb2** [comment]
I don't see errors on the master branch with 6.2.0

```
[~/s/purdy]$ git pull
Already up-to-date.
[~/s/purdy]$ node --version
v6.2.0
[~/s/purdy]$ npm test

> purdy@2.2.0 test /Users/daniel/scripts/purdy
> lab -a code -t 100 -v -I Reflect

Purdy
  ✔ 1) should indent array correctly (15 ms)
  ✔ 2) allows for removal of array index (3 ms)
  ✔ 3) honors indent level (2 ms)
  ✔ 4) prints arguments function (1 ms)
  ✔ 5) should print plain (1 ms)
  ✔ 6) should handle circular references (3 ms)
  ✔ 7) should print a string (0 ms)
  ✔ 8) should print a date (2 ms)
  ✔ 9) should print a number (0 ms)
  ✔ 10) should print null (1 ms)
  ✔ 11) should print undefined (1 ms)
  ✔ 12) should print a regular expression (0 ms)
  ✔ 13) should print a false boolean (1 ms)
  ✔ 14) should print a true boolean (1 ms)
  ✔ 15) should print an empty array without extra indentation (0 ms)
  ✔ 16) should print an empty object without extra indentation (1 ms)
  ✔ 17) should print a more complex object (2 ms)
  ✔ 18) should not print circular after second use (1 ms)
  ✔ 19) shows circular reference with path (1 ms)
  ✔ 20) will keep a path for an object in Hoek format (3 ms)
  ✔ 21) indents object the way it should (1 ms)
  ✔ 22) aligns object the way it should (1 ms)
  ✔ 23) prints directly to console (6 ms)
  errors
    ✔ 24) should display an error (1 ms)
    ✔ 25) should display an error with no error message and property correctly (4 ms)
    ✔ 26) should display an error with properties (1 ms)
    ✔ 27) should display an error with detail (1 ms)
  functions
    ✔ 28) should print constructor name (1 ms)
    ✔ 29) should print not print common constructor (2 ms)
    ✔ 30) should print a function (0 ms)
    ✔ 31) should print an anonymous function (1 ms)
    ✔ 32) should print properties for functions (1 ms)
    ✔ 33) should print properties for functions with name (1 ms)
  symbols
    ✔ 34) prints symbols (1 ms)
    ✔ 35) prints only symbols (1 ms)
    100% coverage
      ✔ 36) should have coverage for having getOwnPropertySymbols (0 ms)
      - 37) should have coverage for not having getOwnPropertySymbols (0 ms)
  depth
    ✔ 38) should handle depth printing (1 ms)
    ✔ 39) should handle depth printing for array (0 ms)
    ✔ 40) should print using zero depth (1 ms)
    ✔ 41) should handle depth printing for mixed (1 ms)
    ✔ 42) should handle depth printing using null depth (1 ms)


41 tests complete
Test duration: 73 ms
Assertions count: 44 (verbosity: 1.07)
No global variable leaks detected
Coverage: 100.00%
```

**Comment by mightyiam** [comment]
@danielb2 then most likely Node.js 6.2.0 has an earlier v8, that doesn't have ES2015 function name inference.

Please read the code changes themselves and the Travis CI logs.

**Comment by danielb2** [comment]
ok, got it. thanks. I'll try to think of a solution for the coverage

**Comment by danielb2** [comment]
You can add a test:

``` javascript
var hayek = function hayek () { this.moop = 3 }
delete hayek.name
```

**Comment by mightyiam** [comment]
Would that be a solution for the coverage? I didn't look into that at all.

The way I see it, if this PR solves some of it properly then just merge it. I can look into the coverage issue afterwards.

**Comment by danielb2** [comment]
Nothing that fails CI is merged. As mentioned, you can add a test containing the hayek stuff there for the coverage

**Comment by danielb2** [comment]
I'm suggesting you take one of the tests:

``` javascript
 it('should print not print common constructor', function (done) {

            var mises = function () { this.moop = 3 }
            delete mises.name;
            var obj = { instance: new mises() };
            var out = Purdy.stringify(obj, { indent: 1 });
            expect(out).to.equal('{\n \u001b[1m\u001b[37minstance\u001b[39m\u001b[22m: {\n  \u001b[1m\u001b[37mmoop\u001b[39m\u001b[22m: \u001b[1m\u001b[34m3\u001b[39m\u001b[22m\n }\n}');
            done();
        });
```

File: test/purdy.js [hunk 1]
```diff
@@ -126,7 +138,12 @@ describe('Purdy', function () {
             var mises = function () { this.moop = 3 }
             var obj = { instance: new mises() };
             var out = Purdy.stringify(obj, { indent: 1 });
-            expect(out).to.equal('{\n \u001b[1m\u001b[37minstance\u001b[39m\u001b[22m: {\n  \u001b[1m\u001b[37mmoop\u001b[39m\u001b[22m: \u001b[1m\u001b[34m3\u001b[39m\u001b[22m\n }\n}');
+            var expected = [
+                '{\n \u001b[1m\u001b[37minstance\u001b[39m\u001b[22m: ',
+                funcNameInfer ? '\u001b[32mmises\u001b[39m ' : '',
+                '{\n  \u001b[1m\u001b[37mmoop\u001b[39m\u001b[22m: \u001b[1m\u001b[34m3\u001b[39m\u001b[22m\n }\n}'
+            ].join('');
```
Comment (danielb2) [inline test/purdy.js]: I've been thinking Is there any way we can do this in a cleaner way?

For instance:
``` javascript
var inferred = funcNameInfer ? '\u001b[32mmises\u001b[39m ' : '',
var expected = '{\n \u001b[1m\u001b[37minstance\u001b[39m\u001b[22m: ' + inferred + '{\n  \u001b[1m\u001b[37mmoop\u001b[39m\u001b[22m: \u001b[1m\u001b[34m3\u001b[39m\u001b[22m\n }\n}'

**Comment by mightyiam** [comment]
The correctness of it is more important, in my eyes, than less lines.

Tests should, the way I see it, test one thing.

BTW, I put quite a bit of thought into this issue and wrote [an article about it](https://medium.com/@mightyiam/when-code-coverage-varies-across-javascript-engines-17a41987f836#.lpvstx6s2).

**Comment by danielb2** [comment]
It's not a question of less lines, it's a question of readability.  I want code to be easily understood and thus easily to maintain.

Nice article :)

**Comment by danielb2** [comment]
thank you

**Comment by danielb2** [comment]
@mightyiam  this is what I meant, FYI: https://github.com/danielb2/purdy.js/commit/c4810a013d4880a527d8dab6e71d171e7e078a01

**Pull request merged**